src/DotNetWorker.Core/Context/Features/DefaultInputConversionFeature.cs

```diff
@@ -133,5 +187,58 @@ private async ValueTask<ConversionResult> AwaitAndReturnConversionTaskResult(Val
 
             }, targetType);
         }
+
+        /// <summary>
+        /// Returns boolean value indicating whether fallback to registered converters enabled by the converter
+        /// </summary>
+        private bool AreFallbackConvertersEnabled(ConverterContext context)
+        {
+            if (context.Properties.TryGetValue(PropertyBagKeys.EnableFallbackConverters, out var res))
+            {
+                if (res is not null && res.GetType() == typeof(bool))
+                {
+                    return (bool)res;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Returns boolean value indicating whether Target type is supported by the converter
+        /// </summary>
+        private bool IsTargetTypeSupportedByConverter(ConverterProperties converterTypeProperties, Type targetType)
+        {
+            return IsTypeSupportedByConverter(converterTypeProperties, targetType)
+                    || IsJsonDeserializedObjectsSupported(converterTypeProperties.SupportsJsonDeserialization, targetType);
+        }
+
+        private bool IsTypeSupportedByConverter(ConverterProperties converterType, Type targetType)
+        {
+            return converterType.SupportedTypes.Any(a => a.AssemblyQualifiedName == targetType.AssemblyQualifiedName);
+        }
+
+        private bool IsJsonDeserializedObjectsSupported(bool converterSupports, Type targetType)
+        {
+            if (converterSupports == true && IsJsonDeserializedObject(targetType))
+            {
+                return true;
+            }
+            else if (targetType.IsArray && targetType.FullName != typeof(byte[]).FullName)
+            {
+                return converterSupports == true && IsJsonDeserializedObject(targetType.GetElementType());
+            }
+            else if (targetType.IsGenericType)
+            {
+                return converterSupports == true && IsJsonDeserializedObject(targetType.GetGenericArguments().FirstOrDefault());
+            }
+
+            return false;
+        }
+
+        private bool IsJsonDeserializedObject(Type type)
+        { 
+            return !type.GetConstructors().Any(a => a.GetParameters().Any());
```

Comment:
Is this trying to say only types with parameterless contrustors are JSON deserializable? If so, this is not true - types can be deserialized from JSON with parametered constructors. Here is how it works for STJ: https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/immutability?pivots=dotnet-7-0

For Newtonsoft it is the same, but implicit - the attribute is not needed. Essentially, you can use constructors with parameters if the params match the property names (pascal to camel case conversion allowed).

Even then, the type could have a custom JSON converter which constructs the object in any number of ways.

What I am getting at is that it is impossible to identify JSON deserialization support statically, and even at runtime we have no idea what the entire set of JSON converters that may be used by the end `InputConverter`.

Reply:
Removed this logic